Fix -Wimplicit-function-declaration, -Wimplicit-int and
-Wincompatible-pointer-type errors with gcc 14.

```
/home/build/aports/testing/odyssey/src/odyssey-1.3/build/third_party/kiwi/kiwi/options.c: In function 'kiwi_parse_options_and_update_vars':
/home/build/aports/testing/odyssey/src/odyssey-1.3/build/third_party/kiwi/kiwi/options.c:44:21: error: implicit declaration of function 'isspace' [-Wimplicit-function-declaration]
   44 |                 if (isspace(pgoptions[i])) {
      |                     ^~~~~~~
/home/build/aports/testing/odyssey/src/odyssey-1.3/build/third_party/kiwi/kiwi/options.c:11:1: note: include '<ctype.h>' or provide a declaration of 'isspace'
   10 | #include "kiwi.h"
  +++ |+#include <ctype.h>
```

```
home/build/aports/testing/odyssey/src/odyssey-1.3/sources/instance.c: In function 'od_bind_version':
/home/build/aports/testing/odyssey/src/odyssey-1.3/sources/instance.c:56:9: error: type defaults to 'int' in type name [-Wimplicit-int]
   56 |         od_asprintf((const **__restrict) & argp_program_version,
      |         ^~~~~~~~~~~
/home/build/aports/testing/odyssey/src/odyssey-1.3/sources/instance.c:56:21: error: passing argument 1 of 'od_asprintf' from incompatible pointer type [-Wincompatible-pointer-types]
   56 |         od_asprintf((const **__restrict) & argp_program_version,
      |                     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                     |
      |                     const int **
In file included from /home/build/aports/testing/odyssey/src/odyssey-1.3/sources/odyssey.h:23:
/home/build/aports/testing/odyssey/src/odyssey-1.3/sources/util.h:30:49: note: expected 'char ** restrict' but argument is of type 'const int **'
   30 | static inline int od_asprintf(char **__restrict bufp, char *fmt, ...)
      |                               ~~~~~~~~~~~~~~~~~~^~~~
```

--- odyssey-1.3-origin/sources/instance.c
+++ odyssey-1.3/sources/instance.c
@@ -51,9 +51,9 @@
 	       path);
 }
 
-static inline void od_bind_version()
+static inline int od_bind_version()
 {
-	od_asprintf((const **__restrict) & argp_program_version,
+  od_asprintf((char **__restrict) & argp_program_version,
 		    "odyssey (git: %s %s %s)", OD_VERSION_NUMBER,
 		    OD_VERSION_GIT, OD_VERSION_BUILD);
 }
--- odyssey-1.3-origin/third_party/kiwi/kiwi/options.c
+++ odyssey-1.3/third_party/kiwi/kiwi/options.c
@@ -6,6 +6,7 @@
  */
 
 #include <assert.h>
+#include <ctype.h> /* isspace() */
 
 #include "kiwi.h"
 
